### PR TITLE
Add custom tap command

### PR DESCRIPTION
**Pull request by mikesplain**
Cherrypicked some of our custom changes as discussed in https://github.com/MikeMcQuaid/strap/pull/122#issuecomment-339602640.

Happy to change however you'd like!

### CODE REVIEW

**Comment by bval** [comment]
I can see the value in this, but might this make more sense as an enhancement to [homebrew-bundle](//github.com/Homebrew/homebrew-bundle)?

Right now `homebrew-bundle` understands custom taps, but not custom commands. I can see it being valuable to tools other than strap to extend it to understand that. You could generalize the existing support for `homebrew-services` to other custom commands. And if you did so, you wouldn't need to add this into strap given the `Brewfile` support.

**Comment by MikeMcQuaid** [comment]
@mikesplain Just so I can think about the implications and implementation of this more: could you give some examples of what your custom commands in the tap do? If it's on the open internet somewhere: I'd love to look at it. Thanks!

**Comment by mikesplain** [comment]
Absolutely, @MikeMcQuaid. We were big users of Boxen but love the simplicity of Strap. Our team was so used to just being able run boxen to update both overal team configuration and project configuration that I wanted to replace it with something similar. We ship most dependencies via dockerfiles and such with our projects however there are a lot of basic tools every member of our team really want installed. 

Although I’m a huge fan of dotfiles (I have my own fork of your awesome dotfiles) many members of our team do not and need some base level config on their systems. So our workflow is to have staff use our strap that adds the custom tap and custom command for that tap. Then in the future they can simply run `brew Barkly` to update their systems. Hope that makes sense, more examples and details are located in the docs. 

https://github.com/barklyprotects/homebrew-barkly

I certainly don’t want to overly expand the role of strap but this has worked tremendously well for our team, especially coming from boxen.

**Comment by MikeMcQuaid** [comment]
@mikesplain Cool, good to know, thanks. Want to sit on this PR for a few more days and mull this over a bit more.

Out of interest (unrelated to this PR): why do you prefer to have Brewfiles, `setup.sh` etc. outside of the project repos? Just curious from a workflow perspective.

Would be great to get a LICENSE on that repo so people can reuse it. Additionally:
https://github.com/barklyprotects/homebrew-barkly/blob/ae5ffd25bdce19cfed37a050f5728aecaadceffa/lib/init.sh#L4-L8

Given that's copy-pasted from Strap could you note the copyright to comply with the license of this repo? Thanks! ❤️

**Comment by mikesplain** [comment]
@MikeMcQuaid Oh wow, so sorry about that. I've added the License and included your copyright notice. Please let me know if that is not sufficient, certainly did not mean to infringe!

As for
>Out of interest (unrelated to this PR): why do you prefer to have Brewfiles, setup.sh etc. outside of the project repos? Just curious from a workflow perspective.

That's an excellent question. I think it comes from how original members of our team used Boxen to setup everything in one repo (our-boxen).  Our actual code repos are broken into many different parts for micro-services and different languages that get pulled together during compilation.  This certainly is not ideal but it was the path we took as we added new features to our products.  

Given how spread out things are, it's challenging for new staff to find which repos, and thus which dependencies are important to them.  Also, most of our repos are interdependent so we use `homebrew-barkly` to update our teams dependencies across the board, rather than having to update them in every repo.  We try to provide all our teams with the ability to write code throughout the full stack, but try to point them in the right direction with the `projects` hierarchy in `homebrew-barkly`, which then basically automates the usage of `brewfiles` etc. 

That said, I personally wish we shipped the dependencies in each repo and used `hombrew-barkly` to simply iterate over them... 

Anyway, I certainly don't mind waiting or making any changes. This is what made sense to us at the time and I hope it (or something similar) could help others. Thanks!

File: README.md [hunk 1]
```diff
@@ -57,6 +57,8 @@ Instead, to deploy to [Heroku](https://www.heroku.com) click:
 - `WEB_CONCURRENCY`: the number of Unicorn (web server) processes to run (defaults to 3).
 - `STRAP_ISSUES_URL`: the URL where users should file issues (defaults to https://github.com/mikemcquaid/strap/issues/new).
 - `STRAP_BEFORE_INSTALL`: instructions displayed in the web application for users to follow before installing Strap (wrapped in `<li>` tags).
+- `CUSTOM_TAP`: an optional tap to strap, allowing other commands to be setup from strap.sh
```
Comment (MikeMcQuaid) [inline README.md]: `an optional Homebrew tap to tap`

File: README.md [hunk 2]
```diff
@@ -57,6 +57,8 @@ Instead, to deploy to [Heroku](https://www.heroku.com) click:
...
+- `CUSTOM_TAP_COMMAND`: an optional tap command to run after a custom tap is setup. e.g. `brew <command from custom_tap>`
```
Comment (MikeMcQuaid) [inline README.md]: `CUSTOM_COMMAND: an single command that is run after all other stages have complete`

File: bin/strap.sh [hunk 3]
```diff
@@ -324,5 +326,19 @@ if [ -f "$HOME/.Brewfile" ]; then
   logk
 fi
 
+# Setup custom tap
```
Comment (MikeMcQuaid) [inline bin/strap.sh]: Think you can kill the comment as it's pretty self-explantory.

File: bin/strap.sh [hunk 4]
```diff
@@ -324,5 +326,19 @@ if [ -f "$HOME/.Brewfile" ]; then
...
+if [ -n "$CUSTOM_TAP" ]; then
+  log "Tapping custom tap:"
+  brew tap "$CUSTOM_TAP"
+  logk
+fi
+
+# Excute custom tap command
```
Comment (MikeMcQuaid) [inline bin/strap.sh]: Think you can kill the comment as it's pretty self-explantory.

File: bin/strap.sh [hunk 5]
```diff
@@ -324,5 +326,19 @@ if [ -f "$HOME/.Brewfile" ]; then
...
+if [ -n "$CUSTOM_TAP_COMMAND" ]; then
+  log "Executing custom tap command:"
```
Comment (MikeMcQuaid) [inline bin/strap.sh]: `log "Executing custom command $CUSTOM_COMMAND:"`

File: bin/strap.sh [hunk 6]
```diff
@@ -324,5 +326,19 @@ if [ -f "$HOME/.Brewfile" ]; then
...
+  brew $CUSTOM_TAP_COMMAND
```
Comment (MikeMcQuaid) [inline bin/strap.sh]: `$CUSTOM_COMMAND` to make this even more generic

**Comment by MikeMcQuaid** [comment]
@mikesplain Any news on this?

**Comment by mikesplain** [comment]
@MikeMcQuaid Sorry for the delay! Updated and squashed.

**Comment by MikeMcQuaid** [comment]
@mikesplain Pushed some changes of my own. If this still looks all 👍 and useful to you from your perspective then shout and I'll merge as-is.

**Comment by mikesplain** [comment]
Looks great @MikeMcQuaid 👍 from me! Thanks!

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Strap, @mikesplain, you rock!

**Comment by mikesplain** [comment]
Thank you for creating an awesome tool @MikeMcQuaid!

**Comment by MikeMcQuaid** [comment]
@mikesplain Thanks again 😊

**Pull request merged**